foundation/network/src/main/kotlin/com/pingidentity/network/HttpRequest.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.network
+
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.buildJsonObject
+
+/**
+ * Interface for building HTTP requests.
+ *
+ * This interface provides methods to configure all aspects of an HTTP request including
+ * URL, headers, query parameters, cookies, and request body. It supports various HTTP
+ * methods (GET, POST, PUT, DELETE) and body formats (JSON, form data).
+ *
+ * The interface is designed to be implemented by different HTTP client libraries while
+ * maintaining a consistent API across the SDK.
+ *
+ */
+interface HttpRequest {
+
+    /**
+     * The URL of the HTTP request.
+     *
+     * This property can be both read and written. Setting this property configures
+     * the target URL for the request.
+     *
+     */
+    var url: String
+
+    /**
+     * Adds a query parameter to the request URL.
+     *
+     * Query parameters are appended to the URL as key-value pairs. Multiple calls
+     * to this method will accumulate parameters.
+     *
+     * @param name The parameter name (will be URL-encoded).
+     * @param value The parameter value (will be URL-encoded).
+     *
+     */
+    fun parameter(
+        name: String,
+        value: String,
+    )
+
+    /**
+     * Adds a header to the request.
+     *
+     * Headers provide metadata about the request. Multiple calls with the same
+     * header name may either replace or append the value depending on the implementation.
+     *
+     * @param name The header name (case-insensitive per HTTP spec).
+     * @param value The header value.
+     *
+     */
+    fun header(
+        name: String,
+        value: String,
+    )
+
+    /**
+     * Adds multiple cookies to the request.
+     *
+     * Cookies are provided in Set-Cookie header format. The implementation will
+     * parse and properly format them for the Cookie header.
+     *
+     * @param cookies List of cookie strings in Set-Cookie format.
+     *
+     */
+    fun cookies(cookies: List<String>)
+
+    /**
+     * Adds a single cookie to the request.
+     *
+     * The cookie should be in Set-Cookie header format. The implementation will
+     * parse and properly format it for the Cookie header.
+     *
+     * @param cookie Cookie string in Set-Cookie format.
+     *
+     */
+    fun cookie(cookie: String)
+
+    /**
+     * Sets the request body as JSON with POST method.
+     *
+     * This method automatically sets the HTTP method to POST and the Content-Type
+     * to application/json. The JSON object is serialized to a string.
+     *
+     * @param body The JSON object to send. Defaults to an empty object.
+     *
+     * ```
+     */
+    fun body(body: JsonObject = buildJsonObject {})
```

Comment:
This only allows JSON bodies via `kotlinx.serialization.JsonObject` and `form-urlencoded` data. There is no way to supply binary payloads. Suggest add different methods:

```
fun body(contentType: String, bytes: ByteArray)
fun body(contentType: String = "application/json", text: String)
```